src/rpc/blockchain.cpp

```diff
@@ -792,7 +792,8 @@ static RPCHelpMan getblockfrompeer()
         "getblockfrompeer",
         "Attempt to fetch block from a given peer.\n\n"
         "We must have the header for this block, e.g. using submitheader.\n"
-        "Subsequent calls for the same block and a new peer will cause the response from the previous peer to be ignored.\n\n"
+        "Subsequent calls for the same block and a new peer will cause the response from the previous peer to be ignored.\n"
+        "Nodes generally ignore requests for a stale block that they never fully verified, or one that is more than a month old.\n\n"
```

Comment:
This will also disconnect the peer, no? https://github.com/bitcoin/bitcoin/pull/20295#discussion_r764737276

If so, I still think it should be mentioned.

Reply:
Yes, when I checked at the time, we'd disconnect. But I haven't tested or re-checked since then.

Reply:
Afaik there's only one place where we track which blocks we requested: `mapBlocksInFlight`. And afaik that map doesn't get vacuum cleaned. So how would a non-response from a peer lead us to disconnect?

Disclaimer: net_processing is very hard to follow, so I probably missed something.

Reply:
```
                LogPrintf("Timeout downloading block %s from peer=%d, disconnecting\n", queuedBlock.pindex->GetBlockHash().ToString(), pto->GetId());

Reply:
Ah yes, we're also updating `state->vBlocksInFlight`, and that does get checked. I'll take a closer look.

Reply:
Took a closer look. It indeed appears this would disconnect. Updated the doc, but haven't tested it..